solutions/usaco-739.mdx

```diff
@@ -0,0 +1,140 @@
+---
+id: usaco-739
+source: USACO Silver 2017 US Open 
+title: Bovine Genomics
+author: Brad Ma
+---
+
+[Official Editorial (C++)](http://www.usaco.org/current/data/sol_cownomics_silver_open17.html)
+
+## Implementation
+
+**Time Complexity** $\mathcal{O}(N^3)$
+
+<LanguageSection>
+  
+<JavaSection>
+  
+<Warning>
+This implementation uses the set data structure, which has not been covered yet. See [Introduction to Sets/Maps](https://usaco.guide/bronze/intro-sets) for more details about sets.
+</Warning>
+  
+```java
+import java.io.*;
+import java.util.*;
+
+public class BovineGenomics {
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio("cownomics");
+	
+		int n = io.nextInt();
+		int m = io.nextInt();
+
+		int[][] spotted = new int[n][m];
+		for (int x = 0; x < n; x++) {  // read spotted
+			StringBuilder line = new StringBuilder();
+			line.append(io.next());
+
+			for (int y = 0; y < m; y++) {
+				// using base 4 will assign a unique
+				// number to every string
+				// A -> 1, C -> 4, T -> 16, G -> 64
+				if (line.charAt(y) == 'A') {
+					spotted[x][y] = 0;
+				}
+				else if (line.charAt(y) == 'C') {
+					spotted[x][y] = 1;
+				}
+				else if (line.charAt(y) == 'T') {
+					spotted[x][y] = 2;
+				}
+				else {
+					spotted[x][y] = 3;
+				}
```

Comment:
```suggestion
				if (line.charAt(y) == 'A') {
					spotted[x][y] = 0;
				} else if (line.charAt(y) == 'C') {
					spotted[x][y] = 1;
				} else if (line.charAt(y) == 'T') {
					spotted[x][y] = 2;
				} else {
					spotted[x][y] = 3;
				}
```

Reply:
that is one of those things that everybody does differently and qualifies as an optional change

Reply:
IDK if USACO Guide uses this, but according to Oracle, [this ](https://www.oracle.com/java/technologies/javase/codeconventions-statements.html#449)is the proper formatting: